### DISCUSSION TITLE
yiwen-blog-website/docs/native/android/viewgroup_view_window

### DISCUSSION THREAD

**Discussion by giscus** [Hao-yiwen/yiwen-blog-website#249] [Announcements]
# yiwen-blog-website/docs/native/android/viewgroup_view_window

在 Android 开发中，ViewGroup、View 和 Window 是非常重要的概念，分别承担着不同的角色和职责。下面详细介绍它们的作用，以及 Window 的 getDecorView 方法的功能。

https://hao-yiwen.github.io/yiwen-blog-website/docs/native/android/viewgroup_view_window

**Comment by Hao-yiwen** [comment]
在activity中获取WindowManager是和当前activity绑定的WindowManager。而这一般不怎么使用，因为activity中还有其他方式进行布局操作。

WindowManager一般指的是application中的WindowManager。例如给app全局添加debug悬浮球。
